samples/cpp/openni_capture.cpp

```diff
@@ -87,90 +87,48 @@ static float getMaxDisparity( VideoCapture& capture )
 {
-    // set defaut values
-    isColorizeDisp = true;
-    isFixedMaxDisp = false;
-    imageMode = 0;
-
-    retrievedImageFlags[0] = false;
-    retrievedImageFlags[1] = true;
-    retrievedImageFlags[2] = false;
-    retrievedImageFlags[3] = true;
-    retrievedImageFlags[4] = false;
-
     filename.clear();
-    isFileReading = false;
-
-    if( argc == 1 )
+    cv::CommandLineParser parser(argc, argv, "{h help||}{cd|1|}{fmd|0|}{mode|0|}{m|01010|}{r||}");
+    if (parser.has("h"))
+    {
+        printCommandLineParams();
+        exit(0);
+    }
+    isColorizeDisp = (parser.get<int>("cd") != 0);
+    isFixedMaxDisp = (parser.get<int>("fmd") != 0);
+    imageMode = parser.get<int>("mode");
+    int flags = parser.get<int>("m");
+    isFileReading = parser.has("r");
+    if (isFileReading)
+        filename = parser.get<string>("r");
+    if (!parser.check())
     {
+        parser.printErrors();
         help();
+        exit(-1);
     }
-    else
+    if (flags % 100000 == 0)
     {
-        for( int i = 1; i < argc; i++ )
-        {
-            if( !strcmp( argv[i], "--help" ) || !strcmp( argv[i], "-h" ) )
-            {
-                printCommandLineParams();
-                exit(0);
-            }
-            else if( !strcmp( argv[i], "-cd" ) )
-            {
-                isColorizeDisp = atoi(argv[++i]) == 0 ? false : true;
-            }
-            else if( !strcmp( argv[i], "-fmd" ) )
-            {
-                isFixedMaxDisp = atoi(argv[++i]) == 0 ? false : true;
-            }
-            else if( !strcmp( argv[i], "-mode" ) )
-            {
-                imageMode = atoi(argv[++i]);
-            }
-            else if( !strcmp( argv[i], "-m" ) )
-            {
-                string mask( argv[++i] );
-                if( mask.size() != 5)
-                    CV_Error( Error::StsBadArg, "Incorrect length of -m argument string" );
-                int val = atoi(mask.c_str());
-
-                int l = 100000, r = 10000, sum = 0;
-                for( int j = 0; j < 5; j++ )
-                {
-                    retrievedImageFlags[j] = ((val % l) / r ) == 0 ? false : true;
-                    l /= 10; r /= 10;
-                    if( retrievedImageFlags[j] ) sum++;
-                }
-
-                if( sum == 0 )
-                {
-                    cout << "No one output image is selected." << endl;
-                    exit(0);
-                }
-            }
-            else if( !strcmp( argv[i], "-r" ) )
-            {
-                filename = argv[++i];
-                isFileReading = true;
-            }
-            else
-            {
-                cout << "Unsupported command line argument: " << argv[i] << "." << endl;
-                exit(-1);
-            }
-        }
+        cout << "No one output image is selected." << endl;
+        exit(0);
+    }
+    for (int i = 0; i < 5; i++)
+    {
+        retrievedImageFlags[i] = (flags % 10 != 0);
```

Comment:
Your variant will reverse the flags:
- original: `00001 ==> retrievedImageFlags[4] = 1`
- yours: `00001 ==> retrievedImageFlags[0] = 1`